src/SprykerEco/Zed/Adyen/Business/Oms/Mapper/RefundCommandMapper.php

```diff
@@ -0,0 +1,74 @@
+<?php
+
+/**
+ * MIT License
+ * Use of this software requires acceptance of the Evaluation License Agreement. See LICENSE file.
+ */
+
+namespace SprykerEco\Zed\Adyen\Business\Oms\Mapper;
+
+use Generated\Shared\Transfer\AdyenApiAmountTransfer;
+use Generated\Shared\Transfer\AdyenApiRefundRequestTransfer;
+use Generated\Shared\Transfer\AdyenApiRequestTransfer;
+use Generated\Shared\Transfer\OrderTransfer;
+use Orm\Zed\Sales\Persistence\SpySalesOrderItem;
+
+class RefundCommandMapper extends AbstractCommandMapper implements AdyenCommandMapperInterface
+{
+    /**
+     * @param \Orm\Zed\Sales\Persistence\SpySalesOrderItem[] $orderItems
+     * @param \Generated\Shared\Transfer\OrderTransfer $orderTransfer
+     *
+     * @return \Generated\Shared\Transfer\AdyenApiRequestTransfer
+     */
+    public function buildRequestTransfer(array $orderItems, OrderTransfer $orderTransfer): AdyenApiRequestTransfer
+    {
+        $request = new AdyenApiRequestTransfer();
+        $paymentAdyen = $this->reader->getPaymentAdyenByOrderTransfer($orderTransfer);
+        $request->setRefundRequest(
+            (new AdyenApiRefundRequestTransfer())
+                ->setMerchantAccount($this->config->getMerchantAccount())
+                ->setOriginalReference($paymentAdyen->getPspReference())
+                ->setOriginalMerchantReference($paymentAdyen->getReference())
+                ->setModificationAmount($this->createAmountTransfer($orderItems, $orderTransfer))
+        );
+
+        return $request;
+    }
+
+    /**
+     * @param \Orm\Zed\Sales\Persistence\SpySalesOrderItem[] $orderItems
+     * @param \Generated\Shared\Transfer\OrderTransfer $orderTransfer
+     *
+     * @return \Generated\Shared\Transfer\AdyenApiAmountTransfer
+     */
+    protected function createAmountTransfer(array $orderItems, OrderTransfer $orderTransfer): AdyenApiAmountTransfer
+    {
+        return (new AdyenApiAmountTransfer())
+            ->setValue($this->getAmountToModify($orderItems, $orderTransfer))
+            ->setCurrency($orderTransfer->getCurrencyIsoCode());
+    }
+
+    /**
+     * @param \Orm\Zed\Sales\Persistence\SpySalesOrderItem[] $orderItems
+     * @param \Generated\Shared\Transfer\OrderTransfer $orderTransfer
+     *
+     * @return int
+     */
+    protected function getAmountToModify(array $orderItems, OrderTransfer $orderTransfer): int
```

Comment:
I see this method repeated. Maybe it should be in an entity of its own.

Reply:
They are different inside.